ChronoVisor/include/ChronicleMetaDirectory.h

```diff
@@ -41,15 +46,11 @@ class ChronicleMetaDirectory {
     std::vector<std::string> show_stories(std::string& client_id, const std::string& chronicle_name);
 
 private:
-//    std::shared_ptr<std::unordered_map<std::string, Chronicle *>> chronicleMap_;
     std::unordered_map<uint64_t , Chronicle *> *chronicleMap_;
-//    std::unordered_map<uint64_t, Chronicle *> *acquiredChronicleMap_;
-    std::unordered_map<uint64_t, Story *> *acquiredStoryMap_;
-    std::unordered_multimap<uint64_t, std::string> *acquiredChronicleClientMap_;
     std::unordered_multimap<uint64_t, std::string> *acquiredStoryClientMap_;
     std::mutex g_chronicleMetaDirectoryMutex_;
-    std::mutex g_acquiredChronicleMapMutex_;
-    std::mutex g_acquiredStoryMapMutex_;
+    std::mutex g_acquiredStoryClientMapMutex_;
+    ClientRegistryManager *clientRegistryManager_{};
 //    std::unordered_map<std::string, uint64_t> *chronicleName2IdMap_;
 //    std::unordered_map<uint64_t, std::string> *chronicleId2NameMap_;
 };
```

Comment:
> Is there a reason that the internal maps are allocated on the heap : pointers to maps instead of just being default constructed members of the ChronicleMetaDirectory? Also exposing these pointers does not seem like a good idea ( lines 25, 49, 50) and initialization in line 53 does not look right , you probably meant to use nullptr, though it's not really needed ,

They were just regularly constructed member objects. But it failed to work when a separate thread is created to run the RPC routine. For some reason, they got overwritten somewhere between initialization here and start of RPC routine code. I can try to reverse it back since Aparna changed the rpc framework a bit. Maybe it is working now.

Reply:
I suspect the issue were the fact that Singleton object used was not protected with the static mutex 
we'll have to revisit this later